src/main/java/com/sprint/mission/discodeit/service/basic/BasicUserService.java

```diff
@@ -48,104 +47,77 @@ public User create(List<MultipartFile> multipartFile, UserCreateRequest userCrea
             throw new DuplicateFormatFlagsException("Username already exists");
         }
 
-        //유저 중복 확인후 유저 생성
-        User user = userRepository.save(new User(
-                        userCreateRequest.username(),
-                        userCreateRequest.email(),
-                        userCreateRequest.password()
-                )
-        );
-
         //파일 있으면 파일 생성
-        if (!multipartFile.isEmpty()) {
-            FileDTO profile = fileUpload.upload(multipartFile).get(0);
-
-            BinaryContent binaryContent = binaryContentRepository.save(
-                    new BinaryContent(
-                            profile.FileName(),
-                            profile.file().getFreeSpace(),
-                            profile.file().getClass().getTypeName(),
-                            profile.file().toString().getBytes(StandardCharsets.UTF_8)
-                    )
-            );
-
-            user.update(UpdateUserDTO.getFileInput(binaryContent.getId()));
+        BinaryContentSave profileBinaryContent = getBinaryContent(multipartFile);
+        BinaryContent profile = null;
+        if (profileBinaryContent != null) {
+            profile = profileBinaryContent.binaryContent();
+            binaryContentStorage.put(profile.getId(), profileBinaryContent.data());
         }
-        userStatusRepository.save(new UserStatus(user.getId()));
-        return userRepository.save(user);
-    }
 
-    @Override
-    public FindUserDTO find(UUID id) {
-        User user = userRepository.findById(id)
-                .orElseThrow(() -> new NoSuchElementException("User not found"));
-        UserStatus userStatus = userStatusRepository.findByUserId(user.getId())
-                .orElseThrow(() -> new NoSuchElementException("UserStatus not found"));
-        return new FindUserDTO(user, userStatus);
-    }
+        //유저 생성
+        User user = new User(
+                userCreateRequest.username(),
+                userCreateRequest.email(),
+                userCreateRequest.password(),
+                profile
+        );
 
-    @Override
-    public FindUserDTO findEmail(String email) {
-        User user = userRepository.findByEmail(email)
-                .orElseThrow(() -> new NoSuchElementException("User not found"));
-        UserStatus userStatus = userStatusRepository.findByUserId(user.getId())
-                .orElseThrow(() -> new NoSuchElementException("UserStatus not found"));
-        return new FindUserDTO(user, userStatus);
+        //유저에 유저 상태 추가
+        UserStatus userStatus = new UserStatus(user);
+        user.update(UpdateUserDto.getStatus(userStatus));
+
+        if (profile != null) {
+            binaryContentRepository.save(profile);
+        }
+        userRepository.save(user);
+        userStatusRepository.save(userStatus);
+        return UserMapper.INSTANCE.toDto(user);
     }
 
     @Override
-    public List<FindUserDTO> findAll() {
+    public List<UserDto> findAll() {
```

Comment:
`readOnly = true` 설정 추가로 성능상의 이점을 얻을 수 있을 것 같아요. 

```suggestion
    @Transactional(readOnly = true)
    public List<UserDto> findAll() {
```